test.js

```diff
@@ -59,6 +59,30 @@ test('stops performing checks if a timeout occurs', async t => {
 		});
 });
 
+test('stops performing async checks if a timeout occurs', async t => {
+	let checksPerformed = 0;
+
+	await pWaitFor(
+		async () => {
+			await new Promise(resolve => {
+				setTimeout(resolve, 20);
+			});
+			checksPerformed += 1;
+			return false;
+		},
+		{
+			interval: 10,
+			timeout: 200,
+		},
+	)
+		.catch(async _ => {
```

Comment:
I would prefer a try/catch instead of `.catch()`. Otherwise, looks good.

Reply:
That entire test was copied from the previous one on line 42, so I tried to stay in the same style. But sure, I can change to try/catch.